modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTransactionalCache.java

```diff
@@ -627,6 +652,177 @@ else if (!F.isEmpty(req.keys()))
         }
     }
 
+    /** {@inheritDoc} */
+    @Override public void unlockAllForSavepoint(@Nullable Collection<IgniteTxKey> keys, GridCacheVersion cacheVer,
```

Comment:
may be break into smaller methods to eliminate duplication and make method more readable